upgrades/upgrade_menu.gd

```diff
@@ -0,0 +1,25 @@
+extends Control
+class_name UpgradeMenu
+
+@onready var upgrades_grid: GridContainer = %UpgradesGrid
+@onready var close_btn: Button = %CloseButton
+const scene = preload("res://upgrades/upgrade_menu.tscn")
+
+static func create() -> UpgradeMenu:
+	var instance = scene.instantiate()
+	return instance
+
+func _ready() -> void:
+	Input.set_mouse_mode(Input.MOUSE_MODE_VISIBLE)
+	close_btn.pressed.connect(_on_close)
+	var upgrades = UpgradeDefinition.get_all()
+	for upgrade in upgrades:
+		var upgrade_item = UpgradeMenuButton.create(upgrade)
+		upgrades_grid.add_child(upgrade_item)
+
+func _on_close() -> void:
+	SignalBus.close_upgrade_menu.emit()
+	self.queue_free()
```

Comment:
Left the comment in the code as well, but it might be better to hide/show this scene instead of creating it and freeing it